### PR TITLE
refactor(kubernetes): refactor Kubernetes provider

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

Refactor code from Kubernetes provider to ensure that it can be used as a library from the SDK.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/kubernetes/kubernetes_provider.py [hunk 1]
```diff
@@ -33,23 +31,28 @@ class KubernetesProvider(Provider):
     # TODO: this is not optional, enforce for all providers
     audit_metadata: Audit_Metadata
 
-    def __init__(self, arguments: Namespace):
+    def __init__(self, kubeconfig_file, context, namespace, config_file, fixer_config):
```
Comment (jfagoagas) [inline prowler/providers/kubernetes/kubernetes_provider.py]: Do we need default values here?

File: prowler/providers/kubernetes/kubernetes_provider.py [hunk 2]
```diff
@@ -33,23 +31,28 @@ class KubernetesProvider(Provider):
...
             logger.info("Retrieving all namespaces ...")
             self._namespaces = self.get_all_namespaces()
         else:
-            self._namespaces = arguments.namespace
+            self._namespaces = namespace
 
         if not self._session.api_client:
             logger.critical("Failed to set up a Kubernetes session.")
-            sys.exit(1)
+            raise SystemExit
```
Comment (jfagoagas) [inline prowler/providers/kubernetes/kubernetes_provider.py:63]: Add a TODO here to replace it with our custom exceptions once they are ready.

File: prowler/providers/kubernetes/kubernetes_provider.py [hunk 3]
```diff
@@ -237,7 +236,7 @@ def search_and_save_roles(
             logger.critical(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
-            sys.exit(1)
+            raise error
```
Comment (jfagoagas) [inline prowler/providers/kubernetes/kubernetes_provider.py:249]: Add a TODO here to replace it with our custom exceptions once they are ready.

File: prowler/providers/kubernetes/kubernetes_provider.py [hunk 4]
```diff
@@ -271,7 +270,7 @@ def get_context_user_roles(self):
...
```
Comment (jfagoagas) [inline prowler/providers/kubernetes/kubernetes_provider.py:284]: Add a TODO here to replace it with our custom exceptions once they are ready.

File: prowler/providers/kubernetes/kubernetes_provider.py [hunk 5]
```diff
@@ -289,7 +288,7 @@ def get_all_namespaces(self) -> list[str]:
             logger.critical(
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
-            sys.exit()
+            raise error
```
Comment (jfagoagas) [inline prowler/providers/kubernetes/kubernetes_provider.py:303]: Add a TODO here to replace it with our custom exceptions once they are ready.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py]: Yes, I added them and the type

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py:63]: Done

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py:249]: Done

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 4 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py:284]: Done

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 5 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py:303]: Done

Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py:303]: Done

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 4 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py:284]: Done

File: prowler/providers/kubernetes/kubernetes_provider.py (hunk 3 above)
Comment (pedrooot) [reply prowler/providers/kubernetes/kubernetes_provider.py:249]: Done

**Review by jfagoagas — DISMISSED** [review]
Go go go go 🚀

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**